pkg/cli/createclusterdefaulter.go

```diff
@@ -13,10 +13,11 @@ type CreateClusterDefaulter struct {
 }
 
 // NewCreateClusterDefaulter to instantiate and register defaults.
-func NewCreateClusterDefaulter(skipIPCheck cluster.ControlPlaneIPCheckAnnotationDefaulter) CreateClusterDefaulter {
+func NewCreateClusterDefaulter(skipIPCheck cluster.ControlPlaneIPCheckAnnotationDefaulter, mhcDefaulter cluster.MachineHealthCheckDefaulter) CreateClusterDefaulter {
```

Comment:
nit: no need to do it now, but I suspect the dependencies here will keep growing. Having an "input" struct will make this easier to extend (for example, reduces the amount of changes in unit tests).

Reply:
https://github.com/aws/eks-anywhere/issues/6285